src/mmvc/base/TriggerMap.hx

```diff
@@ -0,0 +1,236 @@
+			return;
+		var list = stringToCommand.get(trigger);
+		if(list != null)
+			list.remove(command);
+	}
+
+	public function dispatchString(trigger:String)
+	{
+		if(stringToCommand == null)
+			return;
+		var list = stringToCommand.get(trigger);
+		if(list != null)
+			for(commandClass in list)
+				invokeCommand(trigger, String, commandClass);
+	}
+
+	public function mapEnumValue(trigger:EnumValue, command:Class<ICommand>)
+	{
+		if(enumValueToCommand == null)
+			enumValueToCommand = new EnumValueMap<EnumValue, Array<Class<ICommand>>>();
+		var list = enumValueToCommand.get(trigger);
+		(list == null) ? enumValueToCommand.set(trigger, [command]) : list.push(command);
+	}
+
+	public function unmapEnumValue(trigger:EnumValue, command:Class<ICommand>)
+	{
+		if(enumValueToCommand == null)
+			return;
+		var list = enumValueToCommand.get(trigger);
+		if(list != null)
+			list.remove(command);
+	}
+
+	public function dispatchEnumValue(trigger:EnumValue)
+	{
+		if(enumValueToCommand == null)
+			return;
+		var triggerClass = Type.getClass(trigger);
+		var list = enumValueToCommand.get(trigger);
+		if(list != null)
+			for(commandClass in list)
+				invokeCommand(trigger, triggerClass, commandClass);
+	}
+
+	public function mapInstance(trigger:{}, command:Class<ICommand>)
+	{
+		if(!isClassInstance(trigger))
+			throw "Trigger " + Std.string(trigger) + " is not an object.";
+		if(instanceToCommand == null)
+			instanceToCommand = new ObjectMap<Dynamic, Array<Class<ICommand>>>();
+		var list = instanceToCommand.get(trigger);
+		(list == null) ? instanceToCommand.set(trigger, [command]) : list.push(command);
+	}
+
+	public function unmapInstance(trigger:{}, command:Class<ICommand>)
+	{
+		if(!isClassInstance(trigger))
+			throw "Trigger " + Std.string(trigger) + " is not an object.";
+		if(instanceToCommand == null)
+			return;
+		var list = instanceToCommand.get(trigger);
+		if(list != null)
+			list.remove(command);
+	}
+
+	public function dispatchInstance(trigger:{})
+	{
+		if(!isClassInstance(trigger))
+			throw "Trigger " + Std.string(trigger) + " is not an object.";
+		if(instanceToCommand == null)
+			return;
+		var triggerClass = Type.getClass(trigger);
+		var list = instanceToCommand.get(trigger);
+		if(list != null)
+			for(commandClass in list)
+				invokeCommand(trigger, triggerClass, commandClass);
+	}
+
+	function invokeCommand(trigger:Dynamic, triggerClass:Class<Dynamic>, commandClass:Class<ICommand>)
+	{
+		if(commandClass == null)
+			return;
+
+		injector.mapValue(triggerClass, trigger, "trigger");
+		var command = injector.instantiate(commandClass);
+		injector.unmap(triggerClass, "trigger");
+		command.execute();
+	}
+
+	function isClass(source:Dynamic):Bool
+	{
+		return Std.is(source, Class);
+	}
+	
+	function isString(source:Dynamic):Bool
+	{
+		return Std.is(source, String);
+	}
+	
+	function isEnumValue(source:Dynamic):Bool
+	{
+		return Reflect.isEnumValue(source);
+	}
+	
+	function isClassInstance(source:Dynamic):Bool
+	{
```

Comment:
Can this be:

```haxe
return Type.getClass(source) != null;
```

It seems to work OK, just wondering if you'd thought of some issue with that.